test/gurka/test_build_admin.cc

```diff
@@ -0,0 +1,128 @@
+#include <gtest/gtest.h>
+
+#include "gurka.h"
+
+#include "baldr/admin.h"
+#include "mjolnir/admin.h"
+#include "mjolnir/adminbuilder.h"
+#include "mjolnir/pbfgraphparser.h"
+
+using namespace valhalla;
+using namespace valhalla::baldr;
+using namespace valhalla::mjolnir;
+
+class AdminTest : public ::testing::Test {
+protected:
+  static gurka::map admin_map;
+
+  static void SetUpTestSuite() {
+    constexpr double gridsize = 10;
+
+    const std::string ascii_map = R"(
+          A-------B-------C
+          |       |       |
+          |       |       |
+          |       |       |
+          F-------E-------D
+    )";
+
+    // To define an administrative boundary, the nodes must form a closed polygon.
+    const gurka::ways ways = {{"ABCDEFA",
+                               {
+                                   {"highway", "primary"},
+                               }},
+                              {"ABEFA",
+                               {
+                                   {"highway", "primary"},
+                               }},
+                              {"BCDEB",
+                               {
+                                   {"highway", "primary"},
+                               }}};
+
+    const gurka::relations relations = {{{{{gurka::way_member, "ABEFA", "outer"}}},
+                                         {
+                                             {"type", "boundary"},
+                                             {"boundary", "administrative"},
+                                             {"admin_level", "4"},
```

Comment:
it seems to me like we should also add the name of the admin. and then build tiles with this admin sqlite. then we can look at the nodes in the graph and say hey what admin does this have it should be X and actually then you can completely avoid the whole cracking open the sql below and verify that the whole pipeline works (or you could keep the sqlite checks below as extra checks too doesnt hurt!)

Reply:
Named the admins. Added some containment tests.